### PR TITLE
chore(config): create codeowners file

### LINKED ISSUE #149: chore(config): create codeowners file
# What is the problem at hand?
As our project gets larger and larger our internal team can't be expected to keep track of all incoming pull requests. Additionally, we want to ensure that the code that enters our codebase is of high quality and isn't reviewed by someone who doesn't understand the codebase. 

# What is the solution
Create a [codeowners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) file. This will allow us to automatically set required reviewers for an issue. 
**Example**
Let's say I was an expert on the docs. With codeowners any issue that touches the `/docs` directory will automatically have me as a reviewers, AND branch protection rules will be setup so that a codeowners must review the PR.

### PR DESCRIPTION
**Pull request by HasithDeAlwis**
# Linked Issue
Closes #149 

# What was changed
- Created a codeowners file to automate adding pull-request reviewers
- Added small blurb on the docs site to explain to contributors what codeowners are

## Summary by CodeRabbit

- **New Features**
	- Introduced a new section in the contribution guidelines explaining the role of CODEOWNERS in the PR review process.

- **Documentation**
	- Updated PR guidelines to include information about CODEOWNERS and reorganized existing content for clarity.

- **Chores**
	- Implemented a new ownership configuration for various directories and files, designating responsible users for specific areas of the codebase.

- **Tests**
	- Updated test specifications to streamline URL references for documentation navigation.
	- Modified navigation behavior in tests to ensure consistency in URL handling.
	- Improved readability and functionality of helper functions related to page navigation.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (HasithDeAlwis) [reply .github/CODEOWNERS]: @MFarabi619 your call on this one

Comment (MFarabi619) [reply .github/CODEOWNERS]: I'll take it for now

File: .github/CODEOWNERS
Comment (MFarabi619) [reply .github/CODEOWNERS:1]: Alignment is messed up for some reason, looks like this:

image

File: apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx
Comment (MFarabi619) [inline apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:123]: You can actually use [docgen](https://fumadocs.vercel.app/docs/headless/mdx/remark-docgen#file-generator) to render the file directly in here, that way they won't have to click that and go into a different tab.

image

Comment (JeremyFriesenGitHub) [reply apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:123]: essentially, adding code like this would into the cg be worthwhile @HasithDeAlwis @MFarabi619:

```
```json doc-gen:file
{
  "file": "../../.github/CODEOWNERS",
    "codeblock": {
    "lang": "ini",
    "meta": "title=\"CODEOWNERS file\""
  }
}
```

**Review by JeremyFriesenGitHub — CHANGES_REQUESTED** [review]
nitpick: add Jowi to some of the codeowners libs. he helped me setup the db and could also potentially help out with some of that other stuff, such as env.

File: apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx
Comment (HasithDeAlwis) [reply apps/docs/src/content/docs/contribution-guidelines/pull-requests.mdx:123]: Ohhh very nice, thank you for this suggestion

**Review by JeremyFriesenGitHub — COMMENTED** [review]
good job overall!

**Comment by HasithDeAlwis** [comment]
> nitpick: add Jowi to some of the codeowners libs. he helped me setup the db and could also potentially help out with some of that other stuff, such as env.

Yea I totally see where you're coming from, but we'll keep this as is for now and moving forward we'll change codeowners and give more access 👍

**Review by JeremyFriesenGitHub — DISMISSED** [review]

**Review by JeremyFriesenGitHub — DISMISSED** [review]

**Review by JeremyFriesenGitHub — DISMISSED** [review]

**Review by JeremyFriesenGitHub — APPROVED** [review]

**Review by MFarabi619 — APPROVED** [review]

**Pull request merged**
